ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/agent/MLAgentExecutor.java

```diff
@@ -70,26 +78,37 @@ public class MLAgentExecutor implements Executable {
     public static final String MESSAGE_HISTORY_LIMIT = "message_history_limit";
 
     private Client client;
+    private SdkClient sdkClient;
     private Settings settings;
     private ClusterService clusterService;
     private NamedXContentRegistry xContentRegistry;
     private Map<String, Tool.Factory> toolFactories;
     private Map<String, Memory.Factory> memoryFactoryMap;
+    private volatile Boolean isMultiTenancyEnabled;
 
     public MLAgentExecutor(
         Client client,
+        SdkClient sdkClient,
         Settings settings,
         ClusterService clusterService,
         NamedXContentRegistry xContentRegistry,
         Map<String, Tool.Factory> toolFactories,
-        Map<String, Memory.Factory> memoryFactoryMap
+        Map<String, Memory.Factory> memoryFactoryMap,
+        Boolean isMultiTenancyEnabled
     ) {
         this.client = client;
+        this.sdkClient = sdkClient;
         this.settings = settings;
         this.clusterService = clusterService;
         this.xContentRegistry = xContentRegistry;
         this.toolFactories = toolFactories;
         this.memoryFactoryMap = memoryFactoryMap;
+        this.isMultiTenancyEnabled = isMultiTenancyEnabled;
+    }
+
+    @Override
+    public void onMultiTenancyEnabledChanged(boolean isEnabled) {
+        this.isMultiTenancyEnabled = isEnabled;
```

Comment:
I remember this setting is static as https://github.com/opensearch-project/ml-commons/blob/570edaf8ca48f08491706175755330bd24075143/plugin/src/main/java/org/opensearch/ml/settings/MLCommonsSettings.java#L305, are we expecting it can be changed dynamically?

Reply:
So I developed this piece of code when we are thinking to change tenancy dynamically. But for now, we are keeping it as a static settings. But still piece of code applied, it doesn't harm. If we in the long run want to turn it to dynamic settings  we won't need to do anything from our end.

Reply:
Pretty sure that as a static setting it will never trigger this method at all. And you have to actually register listeners to get these notifications and I don't see that done.  So it's half implemented. I agree it's harmless here but I'd rather see none or all.

Reply:
That registration happened in this PR: https://github.com/opensearch-project/ml-commons/pull/3307/files

Reply:
Right but you need to add a settings update consumer in the class where you want it to update.  See https://github.com/search?q=repo%3Aopensearch-project%2Fml-commons%20addsettingsupdateconsumer&type=code 

But I never see us register multitenancy (we did at one time on the feature branch but I think I removed it): https://github.com/opensearch-project/ml-commons/blob/570edaf8ca48f08491706175755330bd24075143/plugin/src/main/java/org/opensearch/ml/settings/MLFeatureEnabledSetting.java#L46-L73

Reply:
Yeah, we just need to add it back [here](https://github.com/opensearch-project/ml-commons/pull/2968/files#diff-c18df262e27ed8263e59dcbd7d1589294a7e2c8615b03248a19190d63193194c), that's all. So for now I'll keep the code as it is.